### ISSUE TITLE
[Bug] Deleting one event wipes out all events

### ISSUE THREAD

**Issue by hwansing** [Etar-Group/Etar-Calendar#1214]
I was just doing some testing for reporting a bug. During this testing, I chose to delete a (ONE !) yearly repeated event with all its repeatations, and - guess what - all events from my calendar were wiped out!!! Hrrrr. What the h.... How can I get my events back?

**Comment by hwansing** [comment]
I have synchronisation to e cloud enabled, can data be restored from the cloud somehow?

**Comment by jspricke** [comment]
Hi, can you give some context which software you use to sync your calendars and to where? I have never heard of such a bug and it sounds a bit unlikely that Etar is at fault (but I don't know all the code).
For recovering the data, Etar only views and manipulates the data but it is stored in the Android calendar provider, a SQLite database. You can try to get data from there but it is unlikely that it is still there. I guess the more promising way is some backup of your cloud.

**Comment by hwansing** [comment]
I have a Fairphone3, bought with installed /e/ OS. This calendar is - I guess - the default calendar APP for /e/. The active synchronization is what's the default for /e/ and what was already installed: sync with e.cloud, which has been renamed into murena cloud, AFAIK.

**Comment by jspricke** [comment]
/e/ has it's own fork of Etar: https://gitlab.e.foundation/e/os/calendar
I would propose to open an issue with them and ask if they can recover your data.
I was not able to reproduce this, did you try?

**Comment by hwansing** [comment]
No, I did not try anything. My guess was, that chance for a successful recovery of my data is best, if I don't change anything on my device now. (Similar to a harddisk, where you deleted something by error. Many write accesses would probably overwrite the data in question.)
If /e/ does not use plain etar, it's probably the wrong place for reporting this issue here, yes.

**Comment by MINBL** [comment]
Hi @jspricke, 

I use /e/OS too and in Etar Calendar and /e/Calendar what I see is that you can't delete only an occurence of a serie.
I guess it's your problem @hwansing? That you deleted the whole serie of event?
You can delete current edited event and the next occurences of the event or all the occurences (before and after the current one which is edited)
Is it a bug of is expected?
If it is expected, do you know if there is an improvement already planned to be able to delete only the current occurence of a serie?

**Comment by bricewge** [comment]
From what I read in the source code, not being able to delete a single event in a series, is a workaround to a bug in `CalendarProvider`.
https://github.com/Etar-Group/Etar-Calendar/blob/6dca08e2073ba17c2ff0e569f318f0b65b228b27/app/src/main/res/values/arrays.xml#L245-L248

 But the previous comment is 15 years old, so I hope it's not a bug anymore.
